leader-common/src/main/kotlin/org/apache/amaterasu/leader/common/configuration/ConfigManager.kt

```diff
@@ -68,4 +73,16 @@ class ConfigManager(private val env: String, private val repoPath: String, priva
     fun getActionConfigContent(action: String, path: String = ""): String {
         return getActionConfiguration(action, path).toYaml.toText()
     }
+
+    fun getFileDataSetConfigurations(): Map<String, FileDataSet> {
```

Comment:
Ideally, this should ideally be all in one datasetConfiguration function, we should only know a name for it. Which means we need a base dataset config as well, maybe with a field describing the type for casting?